src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,265 @@
+                multiplier = getnormal(iv, qp, true)
+                return f_value isa Number ? f_value * multiplier : f_value ⋅ multiplier
+            else
+                normal_dotted || return -f_value
+                multiplier = getnormal(iv, qp, false)
+                return f_value isa Number ? f_value * multiplier : f_value ⋅ multiplier
+            end
+        end
+    end
+end
+
+"""
+    function_value_average(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u))
+
+Compute the average of the function value at the quadrature point over interface.
+"""
+function_value_average
+
+"""
+    function_value_jump(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u), normal_dotted::Bool = true)
+
+Compute the jump of the function value at the quadrature point over the interface.
+
+`normal_dotted::Bool` determines whether to use the definition ``\\llbracket v \\rrbracket=v^- -v^+`` if it's `false`, or
+ the definition  ``\\llbracket v \\rrbracket=v^- ⋅ \\vec{n}^- + v^+ ⋅ \\vec{n}^+`` if it's `true`, which is the default.
+
+!!! note
+    If `normal_dotted == true` then the jump of scalar function values is a vector.
+"""
+function_value_jump
+
+"""
+    function_gradient_average(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u))
+
+Compute the average of the function gradient at the quadrature point over the interface.
+"""
+function_gradient_average
+
+"""
+    function_gradient_jump(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u), normal_dotted::Bool = true)
+
+Compute the jump of the function gradient at the quadrature point over the interface.
+
+`normal_dotted::Bool` determines whether to use the definition ``\\llbracket \\vec{v} \\rrbracket=\\vec{v}^- -\\vec{v}^+`` if it's `false`, or
+ the definition  ``\\llbracket \\vec{v} \\rrbracket=\\vec{v}^- ⋅ \\vec{n}^- + \\vec{v}^+ ⋅ \\vec{n}^+`` if it's `true`, which is the default.
+
+!!! note
+    If `normal_dotted == true` then the jump of the gradient vector is a scalar.
+"""
+function_gradient_jump
+
+for (func,                          f_,                 ) in (
+    (:function_value_average,       :function_value,    ),
+    (:function_gradient_average,    :function_gradient, ),
+)
+    @eval begin
+        function $(func)(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u))
+            dof_range_here = dof_range[dof_range .<= length(eachindex(u)) ÷ 2]
+            dof_range_there = dof_range[dof_range .> length(eachindex(u)) ÷ 2]
+            f_value_here = $(f_)(iv, qp, u, dof_range_here; here = true)
+            f_value_there = $(f_)(iv, qp, u, dof_range_there; here = false)
+            fv = iv.face_values
+            result = 0.5 * f_value_here 
+            fv = iv.other_face_values
+            result += 0.5 * f_value_there
+            return result
+        end
+        # TODO: Deprecate this, nobody is using this in practice...
```

Comment:
:new_moon_with_face:

Reply:
Well, I thought if it's deprecated there why not deprecate it here 😂